lib/mnesia/src/mnesia.erl

```diff
@@ -2441,28 +2444,197 @@ select(Cont) ->
+    InitFun = fun(FixedSpec) -> dirty_sel_init(Where,Tab,FixedSpec,NObjects,Type,reverse) end,
+    fun_select_reverse(Tid,Ts,Tab,Spec,LockKind,Tab,InitFun,NObjects,Where,Type).
+
+-doc false.
+fun_select_reverse(Tid, Ts, Tab, Spec, LockKind, TabPat, Init, NObjects, Node, Storage) ->
+    Def = #mnesia_select{tid=Tid,node=Node,storage=Storage,tab=Tab,orig=Spec},
+    case element(1, Tid) of
+	ets ->
+	    select_state(mnesia_lib:db_select_rev_init(ram_copies,Tab,Spec,NObjects),Def);
+	tid ->
+	    select_lock(Tid,Ts,LockKind,Spec,Tab),
+	    Store = Ts#tidstore.store,
+	    do_fixtable(Tab, Store),
+
+	    Written0 = ?ets_match_object(Store, {{TabPat, '_'}, '_', '_'}),
+	    case Written0 of
+		[] ->
+		    %% Nothing changed in the table during this transaction,
+		    %% Simple case get results from [d]ets
+		    select_state(Init(Spec),Def);
+		_ ->
+		    %% Hard (slow case) records added or deleted earlier
+		    %% in the transaction, have to cope with that.
+		    Type = val({Tab, setorbag}),
+		    Written =
+			if Type == ordered_set -> %% Sort stable, in descending order
+				lists:sort(fun(A, B) -> element(1, A) > element(1, B) end, Written0);
+			   true ->
+				Written0
+			end,
+		    FixedSpec = get_record_pattern(Spec),
+		    CMS = ets:match_spec_compile(Spec),
+		    trans_select(Init(FixedSpec),
+				 Def#mnesia_select{written=Written,spec=CMS,type=Type, orig=FixedSpec})
+	    end;
+	_Protocol ->
+	    select_state(Init(Spec),Def)
+    end.
+
+-doc """
+Continue selecting objects.
+
+Selects more objects with the match specification initiated by
+`mnesia:select_reverse/4`.
+
+Notice that any modifying operations, that is, `mnesia:write` or
+`mnesia:delete`, that are done between the `mnesia:select_reverse/4` and
+`mnesia:select_reverse/1` calls are not visible in the result.
+""".
+-spec select_reverse(Cont) -> {[Match], Cont} | '$end_of_table' when
+      Match::term(),
+      Cont::select_continuation().
+select_reverse(Cont) ->
+    case get(mnesia_activity_state) of
+	{?DEFAULT_ACCESS, Tid, Ts} ->
+	    select_cont(Tid,Ts,Cont,reverse);
+	{Mod, Tid, Ts} ->
+	    Mod:select_cont(Tid,Ts,Cont,reverse);
+	_ ->
+	    abort(no_transaction)
+    end.
+
 -doc false.
-select_cont(_Tid,_Ts,'$end_of_table') ->
+select_cont(Tid,Ts,State) ->
+    select_cont(Tid,Ts,State,forward).
+select_cont(_Tid,_Ts,'$end_of_table',_Dir) ->
     '$end_of_table';
-select_cont(Tid,_Ts,State=#mnesia_select{tid=Tid,cont=Cont, orig=Ms})
+select_cont(Tid,_Ts,State=#mnesia_select{tid=Tid,cont=Cont, orig=Ms},Dir)
   when element(1,Tid) == ets ->
     case Cont of
 	'$end_of_table' -> '$end_of_table';
-	_ -> select_state(mnesia_lib:db_select_cont(ram_copies,Cont,Ms),State)
+	_ ->
+            Result = case Dir of
+                forward -> mnesia_lib:db_select_cont(ram_copies,Cont,Ms);
+                reverse -> mnesia_lib:db_select_rev_cont(ram_copies,Cont,Ms)
+            end,
+            select_state(Result,State)
```

Comment:
While I know that tabs and spaces are mixed everywhere, this looks like wrong indentation in the editor (not here on github).

Reply:
sorry about that, this and your other highlights should be fixed now